File: matchjob_app/lib/model/usuario.dart
class Usuario {
  int id;
  String nome;
  String cpfCnpj;
  String tipoPessoa;
  String senha;
  String email;

  Usuario(this.id, this.nome, this.cpfCnpj,this.tipoPessoa,this.email, this.senha);

  Map toJson() => {
    'nome': nome,
    'cpfCnpj': cpfCnpj,
    'tipoPessoa':tipoPessoa,
    'email': email,
    'senha': senha,
  };

  Usuario.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        nome = json['nome'],
        cpfCnpj = json['cpfCnpj'],
        tipoPessoa = json['tipoPessoa'],
        email = json['email'];
}

File: matchjob_backend/src/main/java/com/br/project/matchjob/model/Usuario.java
package com.br.project.matchjob.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "Usuario")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Usuario implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", unique = true, nullable = false)
    private Long id;

    @Column(name = "nome", nullable = false, length = 60)
    private String nome;

    @Column(name = "cpf", length = 11)
    private String cpf;

    @Column(name = "cnpj", length = 18)
    private String cnpj;

    @Column(name = "tipo_pessoa")
    private char tipoPessoa;

    @Column(name = "email", length = 90)
    private String email;

    @Column(name = "senha", length = 40)
    private String senha;

    @ManyToMany()
    @JoinTable(
            name = "usuario_vaga",
            joinColumns = @JoinColumn(name = "fk_usuario"),
            inverseJoinColumns = @JoinColumn(name = "fk_vaga"))
    private List<Vaga> vagas;

}